webapp/src/app/routes/json/json.component.ts

```diff
@@ -22,14 +22,9 @@ export class JsonComponent {
   fromJSON(): void {
     const blockMap: BlockDataMap = {};
     try {
-      const blockList: BlockData[] = JSON.parse(this.json);
+      const blockList: BlockDataJSON[] = JSON.parse(this.json);
```

Comment:
Why do you need `BlockDataJSON`? I thought it worked also with `BlockData`.
If it's for documenting that with `BlockData` is won't have the methods, you can do that in a comment. I don't think a new type is needed.